src/pages/test-report/cat-a-mod2/components/safety-and-balance/safety-and-balance.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { getTests } from '../../../../../modules/tests/tests.reducer';
+import { getCurrentTest } from '../../../../../modules/tests/tests.selector';
+import { StoreModel } from '../../../../../shared/models/store.model';
+
+import { getTestData } from '../../../../../modules/tests/test-data/cat-a-mod2/test-data.cat-a-mod2.reducer';
+import { map } from 'rxjs/operators';
+
+import { getSafetyAndBalanceQuestions }
+  from '../../../../../modules/tests/test-data/cat-a-mod2/safety-and-balance/safety-and-balance.cat-a-mod2.selector';
+import { FaultCountProvider } from '../../../../../providers/fault-count/fault-count';
+import { Observable } from 'rxjs/Observable';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+
+interface ComponentState {
+  safetyAndBalanceDrivingFaultCount$: Observable<number>;
+}
+
+@Component({
+  selector: 'safety-and-balance',
+  templateUrl: 'safety-and-balance.html',
+})
+export class SafetyAndBalanceComponent implements OnInit {
+  componentState: ComponentState;
+
+  constructor(
+    private store$: Store<StoreModel>,
+    public faultCountProvider: FaultCountProvider) {}
+
+  ngOnInit(): void {
+    const currentTest$ = this.store$.pipe(
+      select(getTests),
+      select(getCurrentTest),
+    );
+
+    this.componentState = {
+      safetyAndBalanceDrivingFaultCount$: currentTest$.pipe(
+        select(getTestData),
+        select(getSafetyAndBalanceQuestions),
+        map((safetyAndBalance) => {
+          return this.faultCountProvider
+            .getSafetyAndBalanceFaultCount(TestCategory.EUA2M2, safetyAndBalance)
```

Comment:
I don't think we should be sending in a hard coded category it should grab the category from the store and pass it in.

Reply:
Yup.

Reply:
Sorted. Rebasing and pushing now.